src/vk/vk_util.c

```diff
@@ -22,16 +22,59 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #include <assert.h>
 
+/*
+Returns number of bits set to 1 in (v).
+
+On specific platforms and compilers you can use instrinsics like:
+
+Visual Studio:
+    return __popcnt(v);
```

Comment:
Maybe @DanielGibson can tough ;-)

Reply:
I can, I've got a VS2022 installed in my Windows VM. Buuuut rev_vk was never ported to VS, we don't have build system support for it... ;)

Reply:
I can confirm that code works with mingw, which takes the gcc codepath. I think we can merge this and wait for someone to complain. ;)

Reply:
:-| unfortunately it seems it does bring issue, at least on macOs

`Product:      Yamagi Quake II
Version:      8.11pre
Platform:     Darwin
Architecture: arm64
Compiler:     Apple LLVM 13.1.6 (clang-1316.0.21.2.5)
Signal:       11

Backtrace:
  0   quake2                              0x0000000102fa6a98 printBacktrace + 48
  1   quake2                              0x0000000102fa6cd8 signalhandler + 240
  2   libsystem_platform.dylib            0x00000001a677b4a4 _sigtramp + 56
  3   ???                                 0xffff8002085ba9d8 0x0 + 18446603344951355864
  4   AGXMetal13_3                        0x00000002085ba9d8 AGXMetal13_3 + 3152344
  5   AGXMetal13_3                        0x00000002085ba960 AGXMetal13_3 + 3152224
  6   libMoltenVK.dylib                   0x0000000112503588 _ZN13MVKImagePlane23updateMTLTextureContentER19MVKImageSubresourceyy + 660
  7   libMoltenVK.dylib                   0x0000000112504254 _ZN21MVKImageMemoryBinding16bindDeviceMemoryEP15MVKDeviceMemoryy + 864
  8   libMoltenVK.dylib                   0x00000001124dc1a4 vkBindImageMemory + 248
  9   ref_vk.dylib                        0x0000000108af4a60 image_create + 540
  10  ref_vk.dylib                        0x0000000108ad7900 QVk_CreateImage + 308
  11  ref_vk.dylib                        0x0000000108ad7b0c QVk_CreateColorBuffer + 92
  12  ref_vk.dylib                        0x0000000108ad16ac CreateDrawBuffers + 196
  13  ref_vk.dylib                        0x0000000108acd91c QVk_Init + 4480
  14  ref_vk.dylib                        0x0000000108ae91f4 RE_InitContext + 164

zsh: segmentation fault  ./quake2`

Reply:
what is difference in selected memory type before and after change? Looks as we was just lucky before because selected just first possible memory type and it had no bad flags in it.

What types of memory is available in vulkan_memory_types_show?

Reply:
Oh right it displays only with validation layers but does not work on my mac so if I make it unconditional

`Memory blocks:
   #0: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT
   #1: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT VK_MEMORY_PROPERTY_HOST_COHERENT_BIT VK_MEMORY_PROPERTY_HOST_CACHED_BIT
   #2: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT VK_MEMORY_PROPERTY_HOST_CACHED_BIT
   #3: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT VK_MEMORY_PROPERTY_LAZILY_ALLOCATED_BIT`

Reply:
Could you compare what memory block used before and after change in vkBindImageMemory? 

code has such https://github.com/yquake2/ref_vk/blob/master/src/vk/vk_image.c#L345 
```c
	return image_create(
		&texture->resource, imageInfo,
		/*mem_properties*/ 0,
		/*mem_preferences*/ VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
```

I suppose that used last memory block with `VK_MEMORY_PROPERTY_LAZILY_ALLOCATED_BIT`. 
```
Memory blocks: 
    #0: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT 
    #1: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT 
          VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT 
          VK_MEMORY_PROPERTY_HOST_COHERENT_BIT 
          VK_MEMORY_PROPERTY_HOST_CACHED_BIT 
    #2: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT 
          VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT 
          VK_MEMORY_PROPERTY_HOST_CACHED_BIT 
>> #3: VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT 
          VK_MEMORY_PROPERTY_LAZILY_ALLOCATED_BIT
```

Reply:
@devnexen Could you check with e5f7bf7e82b0158637fc2ab33563e47138a46094 and reverted 41057458e4b17f2b37595cbb525dcf789bf156db ?

Reply:
it no longer segfault.